soc/arm/microchip_mec/mec1501/device_power.c

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2019 Microchip Technology Inc.
+ * Copyright (c) 2016 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys_io.h>
+#include <misc/__assert.h>
+#include <power.h>
+#include <soc.h>
+
+/*
+ * CPU will spin up to DEEP_SLEEP_WAIT_SPIN_CLK_REQ times
+ * waiting for PCR CLK_REQ bits to clear except for the
+ * CPU bit itself. This is not necessary as the sleep hardware
+ * will wait for all CLK_REQ to clear once WFI has executed.
+ * Once all CLK_REQ signals are clear the hardware will transition
+ * to the low power state.
+ */
+/* #define DEEP_SLEEP_WAIT_ON_CLK_REQ_ENABLE */
+#define DEEP_SLEEP_WAIT_SPIN_CLK_REQ		1000
+
+
+/*
+ * Some peripherals if enabled always assert their CLK_REQ bits.
+ * For example, any peripheral with a clock generator such as
+ * timers, counters, UART, etc. We save the enables for these
+ * peripherals, disable them, and restore the enabled state upon
+ * wake.
+ */
+#define DEEP_SLEEP_PERIPH_SAVE_RESTORE
+
+
+/*
+ * Light sleep: PLL remains on. Fastest wake latency.
+ */
+void soc_lite_sleep_enable(void)
+{
+	SCB->SCR &= ~(1ul << 2);
+	PCR_REGS->SYS_SLP_CTRL = MCHP_PCR_SYS_SLP_LIGHT;
+}
+
+/*
+ * Deep sleep: PLL is turned off. Wake is fast. PLL requires
+ * a minimum of 3ms to lock. During this time the main clock
+ * will be ramping up from ~16 to 24 MHz.
+ */
+
+#if defined(CONFIG_SYS_POWER_DEEP_SLEEP_STATES)
+
+void soc_deep_sleep_enable(void)
+{
+	SCB->SCR = (1ul << 2); /* Cortex-M4 SLEEPDEEP */
+	PCR_REGS->SYS_SLP_CTRL = MCHP_PCR_SYS_SLP_HEAVY;
+}
+
+void soc_deep_sleep_disable(void)
+{
+	SCB->SCR &= ~(1ul << 2); /* disable Cortex-M4 SLEEPDEEP */
+}
+
+
+void soc_deep_sleep_wait_clk_idle(void)
+{
+#ifdef DEEP_SLEEP_WAIT_ON_CLK_REQ_ENABLE
+	u32_t clkreq, cnt;
+
+	cnt = DEEP_SLEEP_WAIT_CLK_REQ;
+	do {
+		clkreq = PCR_REGS->CLK_REQ0 | PCR_REGS->CLK_REQ1
+			 | PCR_REGS->CLK_REQ2 | PCR_REGS->CLK_REQ3
+			 | PCR_REGS->CLK_REQ4;
+	} while ((clkreq != (1ul << MCHP_PCR1_CPU_POS)) && (cnt-- != 0));
+#endif
+}
+
+
+/*
+ * Allow peripherals connected to external masters to wake the PLL but not
+ * the EC. Once the peripheral has serviced the external master the PLL
+ * will be turned back off. For example, if the eSPI master requests eSPI
+ * configuration information or state of virtual wires the EC doesn't need
+ * to be involved. The hardware can power on the PLL long enough to service
+ * the request and then turn the PLL back off.  The SMBus and I2C peripherals
+ * in slave mode can also make use of this feature.
+ */
+void soc_deep_sleep_non_wake_en(void)
+{
```

Comment:
I see PLL never shutting off when triggering deep sleep  from BIOS command line.
Better to have this disabled by default until root cause is known